packages/p2p/src/utils/orders.js

```diff
@@ -1,177 +1,273 @@
-            labels.right_send_or_receive = localize('Send');
-            labels.payment_details = localize('Your payment details');
-            labels.contact_details = localize('Your contact details');
-            labels.instructions = localize('Your instructions');
-            labels.result_string = localize('You sold {{offered_amount}} {{offered_currency}}', {
-                offered_amount: order_details.amount_display,
-                offered_currency: order_details.account_currency,
-            });
-        } else {
-            labels.other_party_role = localize('Seller');
-            labels.left_send_or_receive = localize('Send');
-            labels.right_send_or_receive = localize('Receive');
-            labels.payment_details = localize("Seller's payment details");
-            labels.contact_details = localize("Seller's contact details");
-            labels.instructions = localize("Seller's instructions");
-            labels.result_string = localize("You've received {{offered_amount}} {{offered_currency}}", {
-                offered_amount: order_details.amount_display,
-                offered_currency: order_details.account_currency,
-            });
-        }
-    } else if (is_my_ad) {
-        labels.other_party_role = localize('Seller');
-        labels.left_send_or_receive = localize('Send');
-        labels.right_send_or_receive = localize('Receive');
-        labels.payment_details = localize("Seller's payment details");
-        labels.contact_details = localize("Seller's contact details");
-        labels.instructions = localize('Your instructions');
-        labels.result_string = localize("You've received {{offered_amount}} {{offered_currency}}", {
-            offered_amount: order_details.amount_display,
-            offered_currency: order_details.account_currency,
-        });
-    } else {
-        labels.other_party_role = localize('Buyer');
-        labels.left_send_or_receive = localize('Receive');
-        labels.right_send_or_receive = localize('Send');
-        labels.payment_details = localize('Your payment details');
-        labels.contact_details = localize('Your contact details');
-        labels.instructions = localize("Buyer's instructions");
-        labels.result_string = localize('You sold {{offered_amount}} {{offered_currency}}', {
-            offered_amount: order_details.amount_display,
-            offered_currency: order_details.account_currency,
+class ExtendedOrderDetails {
+    constructor(order_details, loginid, server_time) {
+        this.order_details = order_details;
+        this.loginid = loginid;
+        this.server_time = server_time;
+
+        // Assign all original props to this.
+        Object.keys(this.order_details).forEach(key => {
+            this[key] = order_details[key];
         });
     }
 
-    return labels;
-};
+    // Order statuses
+    get is_buy_order() {
+        return this.order_details.type === 'buy';
```

Comment:
We can use constants now! 😁 
```suggestion
        return this.order_details.type === buy_sell.BUY;
```